libtsuba/include/tsuba/file.h

```diff
@@ -39,6 +39,18 @@ KATANA_EXPORT katana::Result<void> FileStat(
 KATANA_EXPORT katana::Result<void> FileStore(
     const std::string& uri, const uint8_t* data, uint64_t size);
 
+/// Copy a slice of a file from one place to another, out-of-core if possible
```

Comment:
qq: the out-of-core-ness would suggest that there should be a max/target "working memory" parameter. No reason to add it now, but is that the correct model of how this operation should work?

Reply:
In general, yes. Depending on copy location (and certainly for local filesystem copy) the data takes a round trip through memory. S3 is an interesting case since the API allows you to copy from one s3 object directly into another (one of the main reasons I didn't consider a working memory here)

Reply:
Maybe drop the reference to out of core then? At most we can allude to the fact that certain backends have remote-copy capabilities or even scope this API to just remote copies (i.e., same backend).

Reply:
I think I'll re-name it `FileRemoteCopy` and update the comment with those restrictions.